soci/soci_index.go

```diff
@@ -334,23 +335,45 @@ func (b *IndexBuilder) Build(ctx context.Context, img images.Image) (*IndexWithM
 	}
 
 	sociLayersDesc := make([]*ocispec.Descriptor, len(manifest.Layers))
-	eg, ctx := errgroup.WithContext(ctx)
+	var wg sync.WaitGroup
+	var layerCount uint64
+	errChan := make(chan error, len(manifest.Layers))
+
 	for i, l := range manifest.Layers {
-		i, l := i, l
-		eg.Go(func() error {
+		wg.Add(1)
+		go func(i int, l ocispec.Descriptor) {
+			defer wg.Done()
 			desc, err := b.buildSociLayer(ctx, l)
 			if err != nil {
-				return fmt.Errorf("could not build zTOC for layer %s: %w", l.Digest.String(), err)
+				if err != errUnsupportedLayerFormat {
+					errChan <- err
+				}
+				return
 			}
+			atomic.AddUint64(&layerCount, 1)
```

Comment:
It seems a skipped layer due to `errUnsupportedLayerFormat` won't increment this `layerCount`, but a skipped layer due to `min-layer-size` will increment it (line 414 returns `nil`). Is this expected?

Reply:
No. Good catch.